src/resolve/resolved-dns-dnssec.c

```diff
@@ -893,6 +895,9 @@ static int dnssec_rrset_verify_sig(
         initialize_libgcrypt(false);
 #endif
 
+        if (!set_contains(algorithms, INT_TO_PTR(rrsig->rrsig.algorithm)))
+                return -EOPNOTSUPP;
```

Comment:
probably should return a slightly different error code here, to make sure this is recognizable as "disabled by admin"